src/mpid/ch4/netmod/ofi/ofi_send.h

```diff
@@ -480,6 +480,12 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_NM_mpi_send(const void *buf, MPI_Aint count,
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_NM_MPI_SEND);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_NM_MPI_SEND);
 
+#ifdef HAVE_GPU
+    if (MPIDI_IS_BUF_TYPE_GPU(buf) && !MPIDI_OFI_global.can_handle_gpu_mem) {
+        mpi_errno =
+            MPIDIG_gpu_send(buf, count, datatype, rank, tag, comm, context_offset, addr, request);
```

Comment:
@hzhou probably the GPU fallback code should be moved to `mpidig`. If I am not mistaken, this is intended to handle whatever case the netmod cannot handle (e.g., gpu memory). However, I am not sure this is appropriate as `mpidig` is currently used for active messages only. For the GPU fallback we don't need to use AM just do the proper staging before passing the buffer back to netmod.

Reply:
I think `MPIDIG_` prefix is appropriate. I agree with you the concern of mixing GPU code with AM code, and GPU fallback and AM are independent components. Ideally, the `generic` folder should be renamed to `am` folder. How about make a `staging` or `gpu` folder and put your code there? The naming is tricky. I like to use `gpu` as it is short and we all know what this is about. Anyway, let's get the structure right first and wory about the naming later.

Reply:
I moved the gpu code to `src/mpid/ch4/gpu`, although I think the best place is `src/mpid/ch4/generic`; my understanding is that we don't want to treat the GPU code as multimod netmod module as XPMEM is for shared memory. Perhaps we can group `am` and `gpu` (and future generic modules) into `generic` and use the naming scheme you suggested earlier (i.e. `MPIDIG_am_xxx` and `MPIDIG_gpu_xxx`).